eva_submission/submission_qc_checks.py

```diff
@@ -173,204 +313,132 @@ def check_all_browsable_files_are_available_in_ftp(self, vcf_files):
-                    errors.pop('annotate_variants')
-                if 'calculate_statistics' in errors and variant_load_error:
-                    self._statistics_job_check_result = "FAIL"
-                elif 'calculate_statistics' in errors and not variant_load_error:
-                    # We can remove the calculate_statistics error because it is covered by variant_load
-                    errors.pop('calculate_statistics')
-                if not errors:
-                    # If there are no more error we can remove the analysis completely
-                    failed_analysis.pop(analysis_accession)
-
-        report = f"""
-                vcf load result: {self._load_vcf_job_check_result}
-                annotation result: {self._annotation_job_check_result}
-                statistics result: {self._statistics_job_check_result}
-                accession import result: {self._acc_import_job_check_result}"""
-        if failed_files:
-            # For the report the variant_load does not needs to be reported because any new run will be done
-            # with the new load_vcf method. Remove the variant_load from the failed files
-            for file_name in failed_files:
-                failed_files[file_name].pop('variant_load', None)
-            report += f"""
-                    Failed Files:"""
-            for file_name, error_txt in failed_files.items():
-                report += f"""
-                        {file_name}: 
-                            {error_txt.get("load_vcf", "")}
-                            {error_txt.get("acc_import", "")}"""
-        if failed_analysis:
-            report += f"""
-                    Failed Analysis:"""
-            for analysis_accession, error_txt in failed_analysis.items():
-                report += f"""
-                        {analysis_accession}: 
-                            {error_txt.get('annotate_variants', "")}
-                            {error_txt.get('calculate_statistics', "")}"""
-        return report
-
-    def _find_log_and_check_job(self, search_unit, log_file_pattern, job_type, failure_dict=None):
-        log_files = glob.glob(os.path.join(self.path_to_logs_dir, log_file_pattern))
-        report_text = ""
-        if log_files:
-            # check if job completed successfully
-            if not self._did_job_complete_successfully_from_log(log_files[0], job_type):
-                report_text += f"{job_type} failed job/step : {self.get_failed_job_or_step_name(log_files[0])}"
-                job_passed = False
-            else:
-                job_passed = True
+                any_vep_run = True
+                logs_to_check = []
+                jobs_to_check = []
+                for file_name in self.analysis_to_file_names[analysis_accession]:
+                    logs_to_check.append(f"pipeline.*{file_name}*.log")
+                    jobs_to_check.append("variant_load")
```

Comment:
The way I've envisioned these method being used is:

- Assuming that the previously and currently run ingestions are equivalent, we would always report the QC based on the currently run ingestion. 
- So now that we run the variant and study stats separately we would report their QC separately. 
- Since they used to be run together they would both query the previous statistics calculation log on top of the specific variant or study specific log
- This will work well as long as we keep exploding the process in small steps
- We might need to think again if we start combining steps

Reply:
That made me realise that the same logs files will be query multiple time and I forgot to cache the results.
I've also added that.

Reply:
Okay, I think it makes sense and matches what the previous logic was doing as well (I thought previously it was checking annotation and stats steps within the pipeline log and so providing more granular results even for the older runs, but that doesn't seem to be the case.)